content/blog/2019-11-13-space-efficient-gc/index.md

```diff
@@ -0,0 +1,100 @@
++++
+title = "Space Efficient Conservative Garbage Collection"
+extra.author = "Yi Jiang"
+extra.bio = """
+  [Yi Jiang](http://www.cs.cornell.edu/~yijiang/) is a 2nd year Ph.D. student interested in computer architecture and systems.
+"""
+latex = true
+
++++
+
+["Space efficient conservative garbage collection"](https://dl.acm.org/citation.cfm?id=155109) describes some inexpensive but useful techniques that can make *conservative garbage collectors* more space efficient.
+Their techniques can reduce pointer misidentification to retain less memory during runtime, and also excess retention due to less information about variable liveness. Their methods can be easily incorporated into any garbage collecting allocator transparently to client programs.
+
+
+# Pointer Misidentification and Solutions od Existing Conservative Garbage Collectors
+
+The goal of a garbage collector is to retain as little memory as it can, subject to the constraint that all memory that will be accessed in the future must be retained. A garbage collector is called **conservative** if it can operate with minimal information about the layout of the client program's data. 
+
+For conservative collectors, the most apparent potential source of excess memory retention is **pointer misidentification** (e.g., misidentify integers as pointers). Here are some examples:
+
+- An integer variable contains the address of a valid but inaccessible object
+- An object that has a pointer pointing to the interior
```

Comment:
Explain a little more about what you mean by "the interior" and why this is a problem.